packages/features/calendars/weeklyview/components/event/Empty.tsx

```diff
@@ -31,38 +25,92 @@ export function EmptyCell(props: EmptyCellProps) {
     startHour: props.startHour,
   });
 
-  // Empty cell won't show it self (be disabled) when
-  // availableslots is passed in, and it's curren time is not part of the available slots
-  const isDisabled =
-    props.availableSlots &&
-    !props.availableSlots[dayjs(props.day).format("YYYY-MM-DD")]?.find(
-      (slot) => slot.start.getTime() === cellToDate.toDate().getTime()
-    );
+  const minuesFromStart =
+    (cellToDate.toDate().getHours() - props.startHour) * 60 + cellToDate.toDate().getMinutes();
+
+  return <Cell topOffsetMinutes={minuesFromStart} timeSlot={cellToDate} />;
+}
+
+type AvailableCellProps = {
+  availableSlots: CalendarAvailableTimeslots;
+  day: GridCellToDateProps["day"];
+  startHour: GridCellToDateProps["startHour"];
+};
+
+export function AvailableCellsForDay({ availableSlots, day, startHour }: AvailableCellProps) {
+  const slotsForToday = availableSlots && availableSlots[dayjs(day).format("YYYY-MM-DD")];
+
+  const slots = useMemo(
+    () =>
+      slotsForToday?.map((slot) => ({
+        slot,
+        topOffsetMinutes: (slot.start.getHours() - startHour) * 60 + slot.start.getMinutes(),
+      })),
+    [slotsForToday, startHour]
+  );
+
+  if (!availableSlots) return null;
+
+  return (
+    <>
+      {slots?.map((slot) => {
+        return (
+          <Cell
+            key={slot.slot.start.toISOString()}
+            timeSlot={dayjs(slot.slot.start)}
+            topOffsetMinutes={slot.topOffsetMinutes}
+          />
+        );
+      })}
+    </>
+  );
+}
+
+type CellProps = {
+  isDisabled?: boolean;
+  topOffsetMinutes?: number;
+  timeSlot: Dayjs;
+};
+
+function Cell({ isDisabled, topOffsetMinutes, timeSlot }: CellProps) {
```

Comment:
So the majority of this component is moved from above, now a reusable component.